qa/tasks/cephfs/test_cephfs_shell.py

```diff
@@ -393,6 +393,21 @@ def test_get_without_target_name(self):
             # if the int() fails then that means there's a valid stat output
             pass
 
+    def test_get_subdir_creation(self):
+        s = 'somedata'
+        self.get_cephfs_shell_cmd_output("mkdir tmp")
+        self.get_cephfs_shell_cmd_output("put - /tmp/dump7", stdin=s)
+        self.get_cephfs_shell_cmd_output("get /tmp/dump7 ./dump7")
+        # if get cmd is creating subdirs on its own then dump7 will be
+        # stored as ./dump7/tmp/dump7 and not ./dump7, therefore
+        # if doing `cat ./dump7` returns non-zero exit code(i.e. 1) then
+        # it implies that no such file exists at that location
+        r = self.mount_a.run_shell("cat ./dump7", cwd=None,
+                                   check_status=False)
+        log.info("cat ./dump7 output:\n{}".format(r.stdout.getvalue()))
+        self.assertEqual(r.returncode, 0,
+                         "file not found at expected location.")
```

Comment:
If, by chance, return code is 2, the error message added here will be incorrect.

Reply:
It's better I remove it then, if `r.returncode` is not 0, it is clear `cat` failed

Reply:
Isn't keeping that `cat` command a good test for this test case?

Reply:
it would be like `self.assertEqual(r.returncode, 0)`, i think its fine if i don't put the error message, assertion error should be good enough to tell the failure reason